app/scripts/modules/core/src/cluster/filter/ClusterFilters.tsx

```diff
@@ -0,0 +1,334 @@
+import React from 'react';
+import { useOnStateChanged } from '@uirouter/react';
+import { compact, uniq, map } from 'lodash';
+
+import { Application } from 'core/application';
+import { ClusterState } from 'core/state';
+import { FilterCheckbox, ISortFilter, digestDependentFilters } from 'core/filterModel';
+import { robotToHuman, useDataSource, useObservable } from 'core/presentation';
+import {
+  buildLabelsMap,
+  labelFiltersToTrueKeyObject,
+  trueKeyObjectToLabelFilters,
+  ILabelFilter,
+} from './labelFilterUtils';
+
+import { poolBuilder } from './clusterDependentFilterHelper.service';
+import { FilterSection } from './FilterSection';
+import LabelFilter from './LabelFilter';
+
+export interface IClusterFiltersProps {
+  app: Application;
+}
+
+interface IClusterHeaders {
+  account: string[];
+  availabilityZone: string[];
+  category: string[];
+  detail: string[];
+  instanceType: string[];
+  providerType: string[];
+  region: string[];
+  stack: string[];
+}
+
+export const ClusterFilters = ({ app }: IClusterFiltersProps) => {
+  const { serverGroups } = app;
+  const { data: serverGroupData, loaded: clustersLoaded } = useDataSource(serverGroups);
+
+  const [tags, setTags] = React.useState(ClusterState.filterModel.asFilterModel.tags);
+  const [sortFilter, setSortFilter] = React.useState<ISortFilter>(ClusterState.filterModel.asFilterModel.sortFilter);
+  const [labelFilters, setLabelFilters] = React.useState<ILabelFilter[]>(
+    trueKeyObjectToLabelFilters(sortFilter.labels),
+  );
+
+  const labelsMap = buildLabelsMap(serverGroupData);
```

Comment:
May want to memoize this for when serverGroupData is large. 
```suggestion
  const labelsMap = useMemo(() => buildLabelsMap(serverGroupData), [serverGroupData]);
```